### DISCUSSION TITLE
how to add a Roundabout tool on iD Editor

### DISCUSSION THREAD

**Discussion by anithaswain-tomtomexternal** [openstreetmap/iD#9973] [Q&A]
How to add a Roundabout tool in iD Editor

**Comment by k-yle** [comment]
You could use the same technique that's shown in the walkthrough for a water tank:
 - draw a triangle 
 - right click the feature and click circularise

**Reply by anithaswain-tomtomexternal** [reply]
so , roundabout is applicable only on closed road ? Can't we create a roundabout out of  a  straight road element? I have a requirement where i have to make a road element to roundabout .
Do we have any tool to draw a circle on OSM which intersects the road element .
Please give some insights on this.

**Reply by Dimitar5555** [reply]
If you follow Kyle's instruction, you can draw it on top of the existing roads. Then you have to connect each side road to the roundabout. At the end select the center node (where the former intersection was) and press X. Then deselect everything, select the same node and press Delete. This also makes it easier to later deal with all of the relations. What you want to do is possible but it will make a little mess.

**Reply by anithaswain-tomtomexternal** [reply]
Thanks for your prompt response. Can I create a custom tool to draw a circle anywhere on a map like we do it for point/line and increase/decrease the size of it .
